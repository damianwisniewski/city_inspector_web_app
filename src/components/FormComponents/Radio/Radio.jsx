import React from 'react'
import PropTypes from 'prop-types'

const Radio = ({ id, label, ...rest }) => (
	<div className='input-wrapper'>
		<input {...rest} type='radio' className='input-wrapper__field' id={id} />
		<label className='input-wrapper__label' htmlFor={id}>
			{label}
		</label>
	</div>
)

Radio.defaultProps = {
	label: '',
}

Radio.propTypes = {
	id: PropTypes.string.isRequired,
	label: PropTypes.string,
}

export default Radio
