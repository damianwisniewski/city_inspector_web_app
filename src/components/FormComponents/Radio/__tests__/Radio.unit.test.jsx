import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Radio from '../Radio'

configure({ adapter: new Adapter() })

describe('<Radio />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let RadioElement
	const RadioText = 'mockText'
	const id = 'mockId'

	test('should render correctly Radio', () => {
		RadioElement = shallow(<Radio id={id}>{RadioText}</Radio>)
		expect(toJson(RadioElement)).toMatchSnapshot()
	})
})
