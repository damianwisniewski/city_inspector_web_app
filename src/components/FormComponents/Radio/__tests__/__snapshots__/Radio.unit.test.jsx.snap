// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render correctly Radio 1`] = `
<div
  className="input-wrapper"
>
  <input
    className="input-wrapper__field"
    id="mockId"
    type="radio"
  >
    mockText
  </input>
  <label
    className="input-wrapper__label"
    htmlFor="mockId"
  />
</div>
`;
